miner/worker.go

```diff
@@ -92,7 +92,7 @@ type Result struct {
 type worker struct {
 	mu sync.Mutex
 
-	agents []Agent
+	agents map[Agent]struct{}
```

Comment:
OK, this change does not affect performance, maintainability or readability in any significant way. Would therefore recommend to skip it. Also, this change is inconsistent with your statement that the purpose of this PR is to calling XEth.Stop when IPC connections end.

You cannot simultaneously claim that is the purpose of the PR and thus the motivation for not doing other, suggested, code improvement but then do such an unrelated (potential) improvement here.